src/components/core/menu/mainMenu/MainMenuRightChainSelector.tsx

```diff
@@ -0,0 +1,74 @@
+import { ReactComponent as IconCheck } from 'assets/icons/check.svg';
+import { FC, useState } from 'react';
+import { DropdownMenu } from 'components/common/dropdownMenu';
+import { buttonStyles } from 'components/common/button/buttonStyles';
+import { carbonEvents } from 'services/events';
+import { cn } from 'utils/helpers';
+
+interface Props {
+  networks: {
+    id: string;
+    name: string;
+    logoUrl: string;
+    isCurrentNetwork: boolean;
+    selectNetwork: () => void;
+  }[];
+}
+
+export const MainMenuRightChainSelector: FC<Props> = ({ networks }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const activeNetwork = networks.find((network) => network.isCurrentNetwork);
+  if (!activeNetwork) return;
+
+  return (
+    <DropdownMenu
+      isOpen={isOpen}
+      setIsOpen={setIsOpen}
+      placement="bottom"
+      className="space-y-2 rounded-[12px] p-8"
+      aria-expanded={isOpen}
+      button={(attr) => (
+        <button
+          {...attr}
+          className={cn(
+            buttonStyles({ variant: 'secondary' }),
+            'relative flex size-40 items-center justify-center p-0'
+          )}
+          onClick={(e) => {
+            setIsOpen(true);
+            carbonEvents.navigation.navNetworkClick(undefined);
+            attr.onClick(e);
+          }}
+        >
+          <img
+            alt={`Select ${activeNetwork.name}`}
+            src={activeNetwork.logoUrl}
+            className="w-20"
+          />
+        </button>
+      )}
+    >
+      {networks.map((network) => {
+        const { id, name, logoUrl, isCurrentNetwork } = network;
+        return (
+          <button
+            key={id}
+            role="menuitem"
+            className={cn(
+              'rounded-6 flex w-full items-center gap-x-10 p-12',
+              isCurrentNetwork ? 'bg-black' : 'hover:bg-black'
+            )}
+            onClick={network.selectNetwork}
+            disabled={isCurrentNetwork}
+          >
+            <img alt={name} src={logoUrl} className="w-20" />
+            <span>{name}</span>
+            <IconCheck
```

Comment:
You can use the global attribute `hidden` to display none : 
```html
<span className="flex-1">{name}</span>
<IconCheck hidden={!isCurrentNetwork} />
```

Reply:
I decided to use invisible instead of display none to keep the dropdown menu the same width for all chains